src/commands/functions/functions-create.js

```diff
@@ -258,8 +302,17 @@ const ensureFunctionDirExists = async function (command) {
  * @param {import('commander').OptionValues} options
  * @param {string} argumentName
  * @param {string} functionsDir
+ * @param {boolean} isEdgeFunc
  */
-const downloadFromURL = async function (command, options, argumentName, functionsDir) {
+// eslint-disable-next-line max-params
+const downloadFromURL = async function (command, options, argumentName, functionsDir, isEdgeFunc) {
+  if (isEdgeFunc) {
+    log(
+      `${NETLIFYDEVWARN}: Downloading an edge function from a URL is currently not supported. Terminating without further action.`,
+    )
+    process.exit(1)
```

Comment:
I think you could use the `error` helper for this: https://github.com/netlify/cli/blob/main/src/utils/command-helpers.js#L179-L191